autoload/fz.vim

```diff
@@ -42,26 +44,28 @@ function! fz#run(...)
     echohl ErrorMsg | echo "vim-fz doesn't work on legacy vim" | echohl None
     return
   endif
-  let s:ctx = get(a:000, 0, {})
-  if type(s:ctx) != type({})
+  let ctx = {
+    \ 'options': get(a:000, 0, {})
+    \ }
+  if type(ctx['options']) != type({})
     echohl ErrorMsg | echo "invalid argument" | echohl None
     return
   endif
-  let typ = get(s:ctx, 'type', 'cmd')
+  let typ = get(ctx['options'], 'type', 'cmd')
   if typ == 'cmd'
-    let $FZ_IGNORE = get(s:ctx, 'ignore', '(^|[\/])(\.git|\.hg|\.svn|\.settings|\.gitkeep|target|bin|node_modules|\.idea|^vendor)$|\.(exe|so|dll|png|obj|o|idb|pdb)$')
-    let fzcmd = get(s:ctx, 'cmd', s:fz_command)
+    let $FZ_IGNORE = get(ctx['options'], 'ignore', '(^|[\/])(\.git|\.hg|\.svn|\.settings|\.gitkeep|target|bin|node_modules|\.idea|^vendor)$|\.(exe|so|dll|png|obj|o|idb|pdb)$')
+    let fzcmd = get(ctx['options'], 'cmd', s:fz_command)
   else
     echohl ErrorMsg | echo "unsupported type" | echohl None
     return
   endif
-  let s:tmp = tempname()
-  let cmd = printf('%s %s %s > %s', &shell, &shellcmdflag, s:quote(fzcmd), s:tmp)
+  let ctx['tmp_result'] = tempname()
+  let cmd = printf('%s %s %s > %s', &shell, &shellcmdflag, s:quote(fzcmd), ctx['tmp_result'])
+  botright new
```

Comment:
I read the vim docs and tried it. If I pass `curwin: bufnr('%')`  it opens the terminal in that buffer instead of new buffer in vim.